WordPress/src/main/java/org/wordpress/android/util/AnalyticsUtils.java

```diff
@@ -28,32 +29,40 @@
      * @param email WordPress.com email address
      */
     public static void refreshMetadata(String username, String email) {
+        AnalyticsMetadata metadata = new AnalyticsMetadata();
+
         SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(WordPress.getContext());
-        int sessionCount = preferences.getInt(AnalyticsTrackerMixpanel.SESSION_COUNT, 0);
-        boolean isUserConnected = AccountHelper.isSignedIn();
-        boolean isWordPressComUser = AccountHelper.isSignedInWordPressDotCom();
-        boolean isJetpackUser = AccountHelper.isJetPackUser();
-        int numBlogs = WordPress.wpDB.getNumBlogs();
-        int versionCode = PackageUtils.getVersionCode(WordPress.getContext());
-        AnalyticsTracker.refreshMetadata(isUserConnected, isWordPressComUser, isJetpackUser, sessionCount, numBlogs,
-                versionCode, username, email);
+
+        metadata.setSessionCount(preferences.getInt(AnalyticsTrackerMixpanel.SESSION_COUNT, 0));
+        metadata.setUserConnected(AccountHelper.isSignedIn());
+        metadata.setWordPressComUser(AccountHelper.isSignedInWordPressDotCom());
+        metadata.setJetpackUser(AccountHelper.isJetPackUser());
+        metadata.setNumBlogs(WordPress.wpDB.getNumBlogs());
+        metadata.setVersionCode(PackageUtils.getVersionCode(WordPress.getContext()));
```

Comment:
Version code is automatically tracked in Nosara [here](https://github.com/Automattic/Automattic-Tracks-Android/blob/aaa640078a85fda70ed976a6f97b0b65ad498ca4/AutomatticTracks/src/main/java/com/automattic/android/tracks/DeviceInformation.java#L82) and Mixpanel [here](https://github.com/mixpanel/mixpanel-android/blob/2bcc2a22f754550b689865d5b1fe5ed5d215c319/src/main/java/com/mixpanel/android/mpmetrics/MixpanelAPI.java#L221) - So we could drop it from our metadata.

I remember we added it here because it was not supported in Mixpanel at some point ref https://github.com/mixpanel/mixpanel-android/issues/113